ghost/recommendations/src/RecommendationService.ts

```diff
@@ -16,22 +21,40 @@ export class RecommendationService {
     repository: RecommendationRepository;
     wellknownService: WellknownService;
     mentionSendingService: MentionSendingService;
-
-    constructor(deps: {repository: RecommendationRepository, wellknownService: WellknownService, mentionSendingService: MentionSendingService}) {
+    settingsService: SettingsService;
+
+    constructor(deps: {
+        repository: RecommendationRepository,
+        wellknownService: WellknownService,
+        mentionSendingService: MentionSendingService,
+        settingsService: SettingsService,
+    }) {
         this.repository = deps.repository;
         this.wellknownService = deps.wellknownService;
         this.mentionSendingService = deps.mentionSendingService;
+        this.settingsService = deps.settingsService;
     }
 
     async init() {
-        await this.updateWellknown();
+        const recommendations = await this.listRecommendations();
+        await this.updateWellknown(recommendations);
     }
 
-    async updateWellknown() {
-        const recommendations = await this.listRecommendations();
+    async updateWellknown(recommendations: Recommendation[]) {
         await this.wellknownService.set(recommendations);
     }
 
+    async updateRecommendationsEnabledSetting(recommendations: Recommendation[]) {
+        const expectedSetting = (recommendations.length > 0).toString();
+        const currentSetting = this.settingsService.read('recommendations_enabled');
+
+        if (currentSetting && currentSetting === expectedSetting) {
+            return;
+        }
+
+        await this.settingsService.edit([{key: 'recommendations_enabled', value: expectedSetting}], {context: {}});
```

Comment:
We should pass in a context of `{internal: true}` here (because we should block writes to this key if it is not an internal write), but I would abstract that away in the wrapper service instead and simplify the methods that are passed to the RecommendationService

Reply:
Thanks! Refactored in https://github.com/TryGhost/Ghost/pull/17975/commits/43d1781e7d86b2a5a50d754729208135dd805f50